submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/parser/ExperimentSpecParser.java

```diff
@@ -158,6 +172,49 @@ private static V1PodTemplateSpec parseTemplateSpec(ExperimentTaskSpec taskSpec,
     container.setEnv(parseEnvVars(taskSpec, experimentSpec.getMeta().getEnvVars()));
     containers.add(container);
     podSpec.setContainers(containers);
+
+    /**
+     * Init Containers
+     */
+    if (!experimentSpec.getEnvironment().isEmbeddedEnvironment()) {
+      Environment environment = getEnvironment(experimentSpec);
+      if (environment != null) {
+        EnvironmentSpec environmentSpec = environment.getEnvironmentSpec();
+        List<V1Container> initContainers = new ArrayList<>();
+        V1Container initContainer = new V1Container();
+        initContainer.setName(environment.getName());
+        initContainer.setImage(environmentSpec.getDockerImage());
+
+        if (environmentSpec.getKernelSpec().getDependencies().size() > 0) {
+          StringBuffer createCommand = new StringBuffer();
+          String condaEnvironmentName =
+              environmentSpec.getKernelSpec().getName();
+
+          createCommand.append("conda create -n " + condaEnvironmentName);
```

Comment:
Oh yeah, I got the conda related implementation. So basically this added a init container for conda dependency. Do you get chance to test this? Is it actually work? 

Also, it is important to make a better debuggability, for example, this hardcoded a conda path to /opt/conda/envs/env/bin. And i'm not sure if write something to `~/.bashrc` is portable enough or not. I'd like to hear thoughts from @sunilgovind. 

Another thing is, this assumes conda is part of Docker image, what if the Conda is not there, and if that happens, how we can give useful information to users if conda is not installed. 

Also, I think we need to check minimum (or maybe maximum) version of conda.

Reply:
> Do you get chance to test this? Is it actually work?

Yes. Please refer https://github.com/apache/submarine/pull/299#issuecomment-644230611 for more details.

> for example, this hardcoded a conda path to /opt/conda/envs/env/bin. And i'm not sure if write something to `~/.bashrc` is portable enough or not. I'd like to hear thoughts from @sunilgovind.

Had come across mutliple ways while working on it and finally used the approach described here - 
https://medium.com/@chadlagore/conda-environments-with-docker-82cdc9d25754

But, yes, we can improve further.

> Another thing is, this assumes conda is part of Docker image, what if the Conda is not there, and if that happens, how we can give useful information to users if conda is not installed.

Yes, even used the examples/test cases with "continuumio/anaconda3" as dockerImage. This question kind of answer my question mentioned in https://github.com/apache/submarine/pull/299#issuecomment-648614756 ? In this case, users will need to build a new image containing both the stuff?

Will think about doing conda existence check..

> Also, I think we need to check minimum (or maybe maximum) version of conda

Why? Can you explain? Do we need to inform users in case version is not as expected?

Reply:
I'd like to leave detailed review of this part to @sunilgovind or @liuxunorg. I'm not a shell expert. 

> Why? Can you explain? Do we need to inform users in case version is not as expected?

The reason is to avoid unexpected incompatible changes across different versions of Conda. It will be fine if we can give a major (minor) version range as recommendation. Doing a check at the runtime can save people' time to troubleshoot these issues.

Reply:
@manirajv06 can you check my comment above?

Reply:
Taken care. Conda version checks are done. It should fall between min and max version (configurable).